doc/design/dist_train/README.md

```diff
@@ -0,0 +1,117 @@
+# Distributed Training Design Doc
+
+## Goal
+
+### Easy To Use
+
+- Switching from local training to distributed training should be seamless.
+- Researcher can do it effortlessly without the help from engineers.
+
+## Dataset On Cloud
+
+Since distributed training happens on the cloud, dataset needs to live on the cloud as well. We will discuss how to use and create a dataset for distributed training.
+
+### Predefined Dataset
+
+Any dataset in `paddle.dataset` package is a predefined dataset. User can get the [**data reader**](../reader) from the dataset:
+
+```python
+reader = paddle.dataset.cifar.train100()
+```
+
+Decorate the `reader` as usual:
+
+```python
+buffered = paddle.reader.buffered(reader, 200)
+batch = paddle.batch(buffered, 128)
+```
+
+And do distributed train:
+
+```python
+paddle.dist_train(cost, batch, ...)
+```
+
+Or do local train:
+
+```python
+paddle.train(cost, batch, ...)
+```
+
+### Custom Dataset
+
+A custom dataset is identified by name. User only needs to know the dataset name to train on the dataset.
+The complexities like where the dataset is stored, how the dataset is split and distributed are hidden.
+
+User will get a data reader when referencing the dataset by name. User can still decorate data reader as usual, but different from the predefined dataset, the final data reader can only be used for `paddle.dist.train`.
+
+```python
+reader = paddle.dist.dataset("imagenet-custom")
+buffered = paddle.reader.buffered(reader, 200)
+batch = paddle.batch(buffered, 128)
+paddle.dist.train(cost, batch, ...)
```

Comment:
I had some thoughts in https://github.com/PaddlePaddle/Paddle/blob/develop/doc/design/api.md#distributed-training, but there is one thing I am not sure -- how should trainer read data. In above design doc, the master process dispatches *tasks* to trainers. A task could be a filename or a segment of a file or whatever. How could the reader of trainer read the task?